websites/I/Instagram/presence.ts

```diff
@@ -1,37 +1,190 @@
 const presence = new Presence({
-	clientId: "547436289960574977"
-});
+		clientId: "547436289960574977"
+	}),
+	timestamp = Math.floor(Date.now() / 1000);
 
 presence.on("UpdateData", async () => {
 	const presenceData: PresenceData = {
-		largeImageKey: "logo"
-	};
-
-	if (document.location.pathname === "/")
-		presenceData.details = "Viewing the homepage";
-	else if (document.location.pathname.startsWith("/stories")) {
-		const [, , pathName] = document.location.pathname.split("/");
-		presenceData.details = "Viewing a story";
-		presenceData.state = pathName;
-	} else if (document.location.pathname.startsWith("/accounts")) {
+			largeImageKey: "logo"
+		},
+		setting = await getSettings(),
+		{ pathname } = window.location,
+		path = pathname.split("/"),
+		profileName = document.querySelector("div.XBGH5 h2");
+
+	if (setting.elapsedTime) presenceData.startTimestamp = timestamp;
+
+	if (!path[1]) presenceData.details = "Viewing the Homepage";
+	else if (pathname.startsWith("/stories")) {
+		const time = document.querySelector("time.BPyeS.Nzb55"),
+			video = document.querySelector("video");
+
+		presenceData.details = setting.privacy
+			? "Viewing a Story"
+			: `Viewing ${path[2]}'s Story`;
+
+		if (time && time.getAttribute("datetime")) {
+			presenceData.state = getDateString(
+				new Date(time.getAttribute("datetime"))
+			);
+		}
+
+		if (!setting.privacy) {
```

Comment:
spare us some checks and run this at the end (aka delete the unwanted props if true)